osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -22,6 +22,11 @@ public abstract class Mod
         public virtual FontAwesome Icon => FontAwesome.fa_question;
 
         /// <summary>
+        /// The type of this mod.
+        /// </summary>
+        public virtual ModType Type => ModType.Special;

```

Comment:
The previous hardcoded methods allow to fix the position and hotkey of mods implicitly, and their order.
Seems an design stuff.